src/mender-update/daemon/states.cpp

```diff
@@ -106,8 +114,26 @@ void IdleState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 void SubmitInventoryState::DoSubmitInventory(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 	log::Debug("Submitting inventory");
 
-	auto handler = [&ctx, &poster](error::Error err) {
+	auto handler = [this, &ctx, &poster](error::Error err) {
 		if (err != error::NoError) {
+#ifdef MENDER_USE_DBUS
+			// If we get a DBus reply error, we should retry the polling with a short interval
+			// to allow the DBus authentication to be retried.
+			if (err.code == error_condition(dbus::ReplyError, dbus::DBusErrorCategory)) {
+				log::Error("Authentication over DBus failed, retrying");
+				poll_timer_.Cancel();
+				poll_timer_.AsyncWait(chrono::seconds(5), [&poster](error::Error err) {
+					if (err != error::NoError) {
+						if (err.code != make_error_condition(errc::operation_canceled)) {
+							log::Error("DBus poll timer caused error: " + err.String());
+						}
+					} else {
+						poster.PostEvent(StateEvent::InventoryPollingTriggered);
+					}
+				});
+			}
```

Comment:
Missing `return`?

Reply:
I don't believe it's needed? Since we have entered the `SubmitInventoryState`, I figured "Failed to submit inventory: ..." should probably be printed as well

Reply:
Yup, after the second look I agree. :+1: I wasn't quite sure even at the first iteration, hence the `?` :) But yes, we should log an error **and** move to the next state in the state machine (i.e. post a failure event).